test/clj/rems/application/test_model.clj

```diff
@@ -274,262 +299,348 @@
+                                                                 :application/events events
+                                                                 :application/state :application.state/submitted})]
+                                (is (= expected-application (apply-events events)))))))
+
+                        (testing "> submitted (no draft saved)"
+                          (let [events (conj events
+                                             {:event/type :application.event/submitted
+                                              :event/time (DateTime. 7000)
+                                              :event/actor "applicant"
+                                              :application/id 1})
+                                expected-application (deep-merge expected-application
+                                                                 {:application/last-activity (DateTime. 7000)
+                                                                  :application/events events
+                                                                  :application/state :application.state/submitted
+                                                                  ;; when there was no draft-saved event, the current and
+                                                                  ;; previous submitted answers must be the same
+                                                                  :application/form {:form/fields [{:field/value "foo"
+                                                                                                    :field/previous-value "foo"}
+                                                                                                   {:field/value "bar"
+                                                                                                    :field/previous-value "bar"}]}})]
+                            (is (= expected-application (apply-events events)))))))
+
+                    (testing "> licenses added"
+                      (let [events (conj events
+                                         {:event/type :application.event/licenses-added
+                                          :event/time (DateTime. 3500)
+                                          :event/actor "handler"
+                                          :application/id 1
+                                          :application/licenses #{33}
+                                          :application/comment "Please sign these terms also"})
+                            expected-application (merge expected-application
+                                                        {:application/last-activity (DateTime. 3500)
+                                                         :application/modified (DateTime. 3500)
+                                                         :application/events events
+                                                         :application/licenses [{:license/id 30
```

Comment:
If nothing about the original three licenses changed, they should not be repeated here. Just append the new license:

```clj
(-> expected-application
    (merge {:application/last-activity (DateTime. 3500)
            :application/modified (DateTime. 3500)
            :application/events events})
    (update :application/licenses conj {:license/id 33
                                        :license/type :attachment
                                        :license/title {:en "en title"
                                                        :fi "fi title"
                                                        :default "non-localized title"}
                                        :license/attachment-id {:en 3301
                                                                :fi 3302
                                                                :default 3300}
                                        :license/attachment-filename {:en "en filename"
                                                                      :fi "fi filename"
                                                                      :default "non-localized filename"}
                                        :license/start (DateTime. 100)
                                        :license/end nil
                                        :license/expired false
                                        :license/enabled true
                                        :license/archived false}))

Reply:
Now it's much shorter.